capabilities/writetarget/write_target.go

```diff
@@ -0,0 +1,477 @@
+			head, err := c.cs.LatestHead(ctx)
+			if err != nil {
+				lggr.Errorw("failed to fetch the latest head", "txID", txID, "err", err)
+				continue
+			}
+
+			if !accepted {
+				// Check acceptance status
+				status, accepted, statusErr := checkAcceptedStatus(ctx)
+				if statusErr != nil {
+					lggr.Errorw("failed to check accepted status", "txID", txID, "err", statusErr)
+					continue
+				}
+
+				if !accepted {
+					lggr.Infow("not accepted yet", "txID", txID, "status", status)
+					continue
+				}
+
+				lggr.Infow("accepted", "txID", txID, "status", status)
+				// Notice: report write confirmation is only possible after a tx is accepted without an error
+				// TODO: [Beholder] Emit 'platform.write-target.WriteAccepted' (useful to source tx hash, block number, and tx status/error)
+
+				// TODO: check if accepted with an error (e.g., on-chain revert)
+				// Notice: this functionality is not available in the current CW/TXM API
+				acceptedWithErr := false
+				if acceptedWithErr {
+					err = c.beholder.ProtoEmitter.EmitWithLog(ctx, builder.buildWriteError(&info, 0, "write error", "accepted with error"))
+					if err != nil {
+						lggr.Errorw("failed to emit write error", "err", err)
+					}
+					// Notice: no return, we continue to check for confirmation (tx could be accepted by another node)
+				}
+			}
+
+			// Check confirmation status (transmission state)
+			state, err := c.targetStrategy.QueryTransmissionState(ctx, info.reportInfo.reportID, info.request)
+			if err != nil {
+				lggr.Errorw("failed to check confirmed status", "txID", txID, "err", err)
+				continue
+			}
+
+			if state == nil {
+				lggr.Infow("not confirmed yet - transmission state NOT visible", "txID", txID)
+				continue
+			}
+
+			// We (eventually) confirmed the report was transmitted
+			// Emit the confirmation message and return
+			lggr.Infow("confirmed - transmission state visible", "txID", txID)
+
+			// Source the transmitter address from the on-chain state
+			info.reportTransmissionState = state
+
+			err = c.beholder.ProtoEmitter.EmitWithLog(ctx, builder.buildWriteConfirmed(&info, head))
+			if err != nil {
+				lggr.Errorw("failed to emit write confirmed", "err", err)
+			}
+			return
+		}
+	}
+}
+
+// traceAttributes returns the attributes to be used for tracing
+func (c *writeTarget) traceAttributes(workflowExecutionID string) []attribute.KeyValue {
+	return []attribute.KeyValue{
+		attribute.String("capability_id", c.ID),
+		attribute.String("capability_type", string(c.CapabilityType)),
+		attribute.String("workflow_execution_id", workflowExecutionID),
+	}
+}
+
+// asEmittedError returns the WriteError message as an (Go) error, after emitting it first
+func (c *writeTarget) asEmittedError(ctx context.Context, e *wt.WriteError, attrKVs ...any) error {
+	// Notice: we always want to log the error
+	err := c.beholder.ProtoEmitter.EmitWithLog(ctx, e, attrKVs...)
+	if err != nil {
+		return fmt.Errorf("failed to emit error: %+w", err)
```

Comment:
Ensure that we are not dropping the original error
```suggestion
		return errors.Join(fmt.Errorf("failed to emit error: %+w", err), e.AsError())
```